### PR TITLE
Reingold Tilford layout and directed graphs

### PR DESCRIPTION
**Pull request by iosonofabio**
DO NOT MERGE YET!

Trying to address #670 for layouting tree-style directed graphs. The current implementation, which needs a bit of testing, addresses the issue of unreachable nodes within weakly connected components by adding directed nodes from each root to each unreachable node within that component. Better than nothing for 0.8 - and we can brainstorm a bit to improve it in 0.9 if we have time.

Will let you know when this needs serious assessment for merging.

### CODE REVIEW

File: src/layout.c [hunk 1]
```diff
@@ -1115,21 +1116,158 @@ int igraph_layout_reingold_tilford(const igraph_t *graph,
...
+  /* the graph is disconnected and directed, add edges within
+     weakly connected components from the root to all nodes that
+     cannot be reached
+     
+     TODO: a better strategy would be nice here
+     */
+  if (igraph_vector_size(proots) != 1 && mode != IGRAPH_ALL) {
+
+    igraph_vector_t newedges;
+    long int no_of_newedges, root;
```
Comment (vtraag) [inline src/layout.c]: `no_of_newedges` is never initialized, and should be set to 0 at some point. This yields a (possible) error later on with memory allocation when initializing the vector.

**Comment by vtraag** [comment]
Sorry, I don't have further time to look at this now, but I just noticed one problem. I'll try to take another look at some later point.

File: src/layout.c [hunk 2]
```diff
@@ -686,6 +687,81 @@ int igraph_layout_lgl(const igraph_t *graph, igraph_matrix_t *res,
...
+        j++;
+      }
+    }
+  }
+
+  igraph_dqueue_destroy(&q);
+  igraph_adjlist_destroy(&allneis);
+  igraph_vector_destroy(&visited);
+  IGRAPH_FINALLY_CLEAN(3);
+}
```
Comment (vtraag) [inline src/layout.c]: There is now no proper return value, you have to return `IGRAPH_SUCCESS`. Doing so solves the out of memory error I believe (this is just a "random" error, depending on what value was returned).

**Comment by iosonofabio** [comment]
thank you Vincent, good catch. I think we can merge now

**Pull request merged**